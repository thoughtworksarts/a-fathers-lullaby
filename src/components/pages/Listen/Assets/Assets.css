.Assets{
  width: 90%;
  margin: 0 auto 118px auto;
  padding-top: calc(4.5em + 10px);
}

.Assets .number-icon {
  width: 15px;
}

.Assets .clock-icon {
  width: 20px;
}

.Assets .audio-player {
  background-color: #f2f3f5;
  border: 3px solid #929292;
  width: 90%;
  position: fixed;
  bottom: 64px;
  left: 50%;
  transform: translateX(-50%);
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0 10px;
}

.Assets .current-title {
  color: #595959;
  margin: 0;
  white-space: nowrap;
}

.Assets .react-audio-player {
  width: 90%;
}

.Assets thead {
  background-color: #929292;
  color: #ffffff;
}

.Assets tbody {
  background-color: #f2f3f5;
  color: #595959;
}

.Assets .loading {
  text-align: center;
  color: #595959;
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translateY(-50%);
  transform: translateX(-50%);
}
