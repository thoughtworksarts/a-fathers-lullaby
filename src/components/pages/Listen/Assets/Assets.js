import React, { useState, useEffect } from 'react'
import Asset from './Asset/Asset'
import Table from 'react-bootstrap/Table'
import './Assets.css'
import ReactAudioPlayer from 'react-audio-player'
import numberIcon from './hashtag-solid.svg'
import clockIcon from './clock-regular.svg'


const Assets = () => {
  const [assets, setAssets] = useState([])
  const [currentStoryIndex, setCurrentStoryIndex] = useState('')
  const [currentTitle, setCurrentTitle] = useState('')
  const [currentFilename, setCurrentFilename] = useState('')

  useEffect(() => {
    fetch(`${process.env.REACT_APP_CORS_ANYWHERE}/${process.env.REACT_APP_ASSETS_URL}`, {
      headers: {
        'authorization': `token ${process.env.REACT_APP_ROUNDWARE_TOKEN}`
      }
    })
      .then(res => res.json())
      .then(assets => setAssets(assets.sort((a, b) => (a.created < b.created) ? 1 : -1)))
      .catch(err => console.log(err))
  }, [])

  const clickHandler = (index, title, filename) => {
    setCurrentStoryIndex(index)
    setCurrentTitle(title)
    setCurrentFilename(filename)
  }

  // TODO: Change to asset id if it's possible to update the current ids to start at 1
  let listNumber = 0

  const endHandler = () => {
    const nextStoryIndex = currentStoryIndex + 1
    
    if (nextStoryIndex === assets.length) {
      return null
    } else {
      setCurrentStoryIndex(nextStoryIndex)
      setCurrentTitle('Story ' + assets[nextStoryIndex].id)
      setCurrentFilename(assets[nextStoryIndex].filename)
    }
  }

  return (
    <div className="Assets">
      {
        assets.length > 0
          ? <React.Fragment>
              <div className="audio-player">
                <p className="current-title">{currentTitle}</p>
                <ReactAudioPlayer
                  src={process.env.REACT_APP_ROUNDWARE_PROD + currentFilename}
                  controls
                  autoPlay
                  onEnded={endHandler}
                />
              </div>
              <Table hover>
                <thead>
                  <tr>
                    <th><img src={numberIcon} className="number-icon" alt="number icon" /></th>
                    <th>Title</th>
                    <th>Location</th>
                    <th><img src={clockIcon} className="clock-icon" alt="clock icon" /></th>
                  </tr>
                </thead>
                <tbody>
                  {
                    assets.map(asset => {
                      listNumber += 1
                      return <Asset key={asset.id} asset={asset} listNumber={listNumber} clickHandler={clickHandler} />
                    })
                  }
                </tbody>
              </Table>
            </React.Fragment>
          : <h1 className="loading">Loading...</h1>
      }
    </div>
  )
}

export default Assets
