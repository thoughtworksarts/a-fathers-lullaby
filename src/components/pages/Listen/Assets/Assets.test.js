import React from 'react'
import Assets from './Assets'
import { shallow } from 'enzyme'

describe('Assets', () => {
  it('matches snapshot', () => {
    const assetsComponent = shallow(<Assets />)
    expect(assetsComponent).toMatchSnapshot()
  })
})
