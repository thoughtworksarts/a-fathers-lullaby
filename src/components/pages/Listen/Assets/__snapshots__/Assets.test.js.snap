// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assets matches snapshot 1`] = `
<div
  className="Assets"
>
  <h1
    className="loading"
  >
    Loading...
  </h1>
</div>
`;
